### PR TITLE
docs: add typings to some examples

### PR DESCRIPTION
**Pull request by dave-meyer**
## Description

The previous code examples on the website sometimes used untyped 'useState' hooks. This made it sometimes difficult to understand the code. My changes add strong typing (i.e. ``useState<Date>()``) to wherever the code was not self explanatory.

Also, these changes unify the code examples, since some of them had strongly typed hooks and others did not.

## Type of Change

- [x] Documentation update

## Checklist

Before submitting your pull request, please make sure the following is done:

- [x] I have read the [CONTRIBUTING](./CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works (not needed)
- [x] I have added necessary documentation (if appropriate)
- [x] I have followed the coding guidelines in this project
- [x] I have tested my changes on different browsers and screen sizes (not needed)

## Linked Issues

None

## Test Plan

Not needed

## Screenshots (if appropriate)

Not needed

## Further Comments

None

### CODE REVIEW

**Comment by dave-meyer** [comment]
I think this PR is ready for review/merge now.

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ import { DayPicker } from "react-day-picker";
 import "react-day-picker/dist/style.css";
 
 function MyDatePicker() {
-  const [selected, setSelected] = useState();
+  const [selected, setSelected] = useState<Date>();
```
Comment (gpbl) [inline README.md:36]: Single mode allows also `undefined` values.

```suggestion
  const [selected, setSelected] = useState<Date | undefined>();
```

File: website/docs/start.mdx [hunk 2]
```diff
@@ -30,7 +30,7 @@ import { DayPicker } from "react-day-picker";
 import "react-day-picker/style.css";
 
 export function MyDatePicker() {
-  const [selected, setSelected] = useState();
+  const [selected, setSelected] = useState<Date>();
```
Comment (gpbl) [inline website/docs/start.mdx:33]:
```suggestion
  const [selected, setSelected] = useState<Date | undefined>();
```

File: website/docs/using-daypicker/navigation.mdx [hunk 3]
```diff
@@ -45,7 +45,7 @@ export function Controlled() {
   const today = new Date();
   const nextMonth = addMonths(today, 1);
 
-  const [month, setMonth] = useState(nextMonth);
+  const [month, setMonth] = useState<Date>(nextMonth);
```
Comment (gpbl) [inline website/docs/using-daypicker/navigation.mdx:48]: Isn't this inferred already?

```suggestion
  const [month, setMonth] = useState(nextMonth);
```

File: website/docs/using-daypicker/selection-modes.mdx [hunk 4]
```diff
@@ -49,7 +49,7 @@ import { DayPicker } from "react-day-picker";
 
 export function App() {
   const initiallySelectedDate = new Date();
-  const [selectedDate, setSelectedDate] = useState(initiallySelectedDate);
+  const [selectedDate, setSelectedDate] = useState<Date>(initiallySelectedDate);
```
Comment (gpbl) [inline website/docs/using-daypicker/selection-modes.mdx:52]: This should also be inferred by the type of `initiallySelectedDate`.

```suggestion
  const [selectedDate, setSelectedDate] = useState(initiallySelectedDate);
```

File: website/docs/using-daypicker/selection-modes.mdx [hunk 5]
```diff
@@ -217,7 +217,7 @@ import { useState } from "react";
 import { DayPicker } from "react-day-picker";
 
 export function App() {
-  const [selected, setSelected] = useState();
+  const [selected, setSelected] = useState<Date | undefined>();
```
Comment (gpbl) [inline website/docs/using-daypicker/selection-modes.mdx:220]: This should be an array.

```suggestion
  const [selected, setSelected] = useState<Date[] | undefined>();
```

**Review by gpbl — CHANGES_REQUESTED** [review]
Thanks a lot for your update. Not all the changes are correct, however - would you mind to fix these? Thanks again.

**Comment by dave-meyer** [comment]
Thank you for reviewing! I commited your requested changes.

**Review by gpbl — APPROVED** [review]

**Comment by gpbl** [comment]
Thanks _you_, @dave-meyer !

**Pull request merged**